kinto/core/resource/__init__.py

```diff
@@ -460,17 +461,42 @@ def patch(self):
         existing = self._get_record_or_404(self.record_id)
         self._raise_412_if_modified(existing)
 
-        try:
-            # `data` attribute may not be present if only perms are patched.
-            changes = self.request.json.get('data', {})
-        except ValueError:
-            # If no `data` nor `permissions` is provided in patch, reject!
-            # XXX: This should happen in schema instead (c.f. ShareableViewSet)
-            error_details = {
-                'name': 'data',
-                'description': 'Provide at least one of data or permissions',
-            }
-            raise_invalid(self.request, **error_details)
+        content_type = str(self.request.headers.get('Content-Type'))
+        # patch is specified as a list of operations (RFC 6902)
+        if content_type == 'application/json-patch+json':
+            try:
+                ops = self.request.json
+                self.patch_changes = JsonPatch(ops, existing)
+
+                if '__permissions__' in existing:
+                    perms = existing['__permissions__']
+                    self.patch_changes.base_permissions.update(perms)
```

Comment:
I know this part is a bit messy. It will be better with #790, but in the meantime **if** you can keep the *permissions* to the children class `ShareableResource` it would be better :) Didn't look in details if feasible for this case though